### PR TITLE
Fix inconsistent base-type of AVX2 shift intrinsics

### PR DESCRIPTION
**Pull request by fiigii**
@tannergooding @eerhardt @jkotas @CarolEidt

### CODE REVIEW

File: src/mscorlib/src/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs [hunk 1]
```diff
@@ -1395,7 +1395,7 @@ public static class Avx2
         /// __m256i _mm256_sllv_epi32 (__m256i a, __m256i count)
         ///   VPSLLVD ymm, ymm, ymm/m256
         /// </summary>
-        public static Vector256<int> ShiftLeftLogicalVariable(Vector256<int> value, Vector256<uint> count) { throw new PlatformNotSupportedException(); }
+        public static Vector256<int> ShiftLeftLogicalVariable(Vector256<int> value, Vector256<int> count) { throw new PlatformNotSupportedException(); }
```
Comment (jkotas) [inline src/mscorlib/src/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs]: Why is it a good idea for the shift amount to be signed here?

Comment (tannergooding) [reply src/mscorlib/src/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs]: The architecture manual actually specifies the second operand is unsigned:
> Shifts the bits in the individual data elements (words, doublewords or quadword) in the first source operand to the
left by the count value of respective data elements in the second source operand. As the bits in the data elements
are shifted left, the empty low-order bits are cleared (set to 0).
>
> The count values are specified individually in each data element of the second source operand. If the unsigned
integer value specified in the respective data element of the second source operand is greater than 15 (for word),
31 (for doublewords), or 63 (for a quadword), then the destination data element are written with 0.

Comment (fiigii) [reply src/mscorlib/src/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs]: Hmm..For this kind of intrinsic (that take count/mask vector), we are always  using same base-type for data and count/mask. But shift* looks special. Do you think unsigned base-type is better? If yes, I will change SSE2 counterparts.

Comment (4creators) [reply src/mscorlib/src/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs]: Actually, despite that the fact that Intel manual specifies that count values are unsigned they significance is limited to 4 bits for a word, 5 bits for a doubleword and 63 bits for a quadword, therefore, sign bit does not matter. It is rather more a API design convention or a developers coding convenience which should influence our decision in such case.

In my experience it is rather very probable that the types used by intrinsics code would the same as first argument and return vector base type.

Comment (fiigii) [reply src/mscorlib/src/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs]: > It is rather more a API design convention or a developers coding convenience which should influence our decision in such case.

> In my experience it is rather very probable that the types used by intrinsics code would be the same as first argument and return vector base type.

Yes, that is also my point.

**Pull request closed without merging**